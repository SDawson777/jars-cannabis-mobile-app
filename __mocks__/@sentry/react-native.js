// Manual Jest mock for @sentry/react-native used during tests.
module.exports = {
  addBreadcrumb: () => {},
  addEventProcessor: () => {},
  captureException: () => {},
  captureEvent: () => {},
  captureMessage: () => {},
  init: () => {},
  setUser: () => {},
  setContext: () => {},
  Scope: function Scope() {},
  withScope: (fn) => fn && fn(),
  lastEventId: () => null,
};
