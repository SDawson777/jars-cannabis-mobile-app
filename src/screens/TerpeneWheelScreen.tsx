import React from 'react';
import { View, Text, StyleSheet, AccessibilityInfo } from 'react-native';

import { TerpeneInfoModal } from '../terpene_wheel/components/TerpeneInfoModal';
import { TerpeneWheel } from '../terpene_wheel/components/TerpeneWheel';
import type { TerpeneInfo } from '../terpene_wheel/data/terpenes';

export default function TerpeneWheelScreen() {
  const [selectedTerpene, setSelectedTerpene] = React.useState<TerpeneInfo | null>(null);

  React.useEffect(() => {
    AccessibilityInfo.announceForAccessibility?.('Terpene Wheel');
  }, []);

  return (
    <View style={styles.container} testID="terpene-wheel-screen">
      <Text style={styles.title}>Terpene Wheel</Text>
      <TerpeneWheel onSelect={setSelectedTerpene} />
      {selectedTerpene && (
        <TerpeneInfoModal
          terpene={selectedTerpene}
          visible={!!selectedTerpene}
          onClose={() => setSelectedTerpene(null)}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingHorizontal: 16,
    paddingVertical: 24,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 16,
  },
});
