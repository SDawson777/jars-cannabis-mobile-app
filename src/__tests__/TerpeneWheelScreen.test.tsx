import { render } from '@testing-library/react-native';
import React from 'react';

import TerpeneWheelScreen from '../screens/TerpeneWheelScreen';

// Mock the terpene wheel components
jest.mock('../terpene_wheel/components/TerpeneWheel', () => ({
  TerpeneWheel: ({ onSelect }: any) => {
    const MockTerpeneWheel = require('react').createElement;
    return MockTerpeneWheel('View', { testID: 'terpene-wheel', onPress: () => onSelect({}) });
  },
}));

jest.mock('../terpene_wheel/components/TerpeneInfoModal', () => ({
  TerpeneInfoModal: () => {
    const MockModal = require('react').createElement;
    return MockModal('View', { testID: 'terpene-info-modal' });
  },
}));

describe('TerpeneWheelScreen', () => {
  it('renders correctly and matches snapshot', () => {
    const component = render(<TerpeneWheelScreen />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders TerpeneWheel component', () => {
    const { getByTestId } = render(<TerpeneWheelScreen />);
    expect(getByTestId('terpene-wheel-screen')).toBeTruthy();
  });

  it('has accessible title', () => {
    const { getByText } = render(<TerpeneWheelScreen />);
    expect(getByText('Terpene Wheel')).toBeTruthy();
  });
});
