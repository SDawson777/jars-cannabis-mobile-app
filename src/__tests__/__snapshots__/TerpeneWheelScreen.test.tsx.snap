// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerpeneWheelScreen renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 24,
    }
  }
  testID="terpene-wheel-screen"
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    Terpene Wheel
  </Text>
  <View
    onPress={[Function]}
    testID="terpene-wheel"
  />
</View>
`;
